pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -87,11 +87,18 @@ func isExecutorTerminated(executorState v1beta1.ExecutorState) bool {
 	return executorState == v1beta1.ExecutorCompletedState || executorState == v1beta1.ExecutorFailedState
 }
 
-func driverPodPhaseToApplicationState(podPhase apiv1.PodPhase) v1beta1.ApplicationStateType {
-	switch podPhase {
+func driverStateToApplicationState(podStatus apiv1.PodStatus) v1beta1.ApplicationStateType {
+	switch podStatus.Phase {
 	case apiv1.PodPending:
 		return v1beta1.SubmittedState
 	case apiv1.PodRunning:
+		// TODO: upcoming Kubernenetes feature will make this code redundant
+		// https://github.com/kubernetes/enhancements/issues/753
+		for _, c := range podStatus.ContainerStatuses {
+			if c.Name == config.SparkDriverContainerName && c.State.Terminated != nil && c.State.Terminated.ExitCode == 0 {
+				return v1beta1.SucceedingState
```

Comment:
You still need to check the case of `c.State.Terminated.ExitCode != 0` and return `v1beta1.FailingState` in that case.

Reply:
If the Terminated.ExitCode != 0 on one of the containers, ist the pod will mark as failed.

Reply:
The `PodPhase` stays `Running` if the sidecar container is still running.

Reply:
Thanks for pointing it out.